app/controllers/leaves_controller.rb

```
@@ -37,17 +37,34 @@ def create
   end
 
   def edit
-    response = @client.edit_leave(params[:calendar_id],params[:id])
-    render locals:{ leave: JSON.parse(response.body) }
+    leave = Leave.find(params[:id])
+    render locals:{ leave: leave, reasons: Reason.all }
   end
 
   def update
+    if(DURATION_TYPE.include?(params[:leave_duration]))
+      start_date = params[:start].to_time(:utc).iso8601
+      end_date = params[:start].to_time(:utc).iso8601
+      leave_type = 1
+      summary = "#{current_user.name} is on Half Day Leave (#{Reason.find_by(id: params[:leave][:reason_id]).name})"
+    else
+      leave_type = 0
+      start_date = params[:start].to_time(:utc).iso8601
+      end_date = params[:end].to_time(:utc).iso8601
+      summary = "#{current_user.name} is on Leave (#{Reason.find_by(id: params[:leave][:reason_id]).name})"
+    end
     leave = {
-      summary: params[:summary],
-      start: {dateTime: params[:start].to_time(:utc).iso8601},
-      end: {dateTime: params[:end].to_time(:utc).iso8601}
+      summary: summary,
+      start: {dateTime: start_date},
+      end: {dateTime: end_date}
     }
-    @client.update_leave(params[:calendar_id], params[:id], leave)
+    calendar_leaves = Calendarleave.where(leave_id: params[:leave_id])
+    calendar_leaves.each do |calendarleave|
+      calendar = Calendar.find(calendarleave.calendar_id)
+      @client.update_leave(calendar.google_id, calendarleave.leave_google_id, leave)
+    end
+    leave_db = Leave.find(params[:leave_id])
```

move finds on params on top as much as possible. want to fail early if the params are invalid. not after running the bigger calendar-leaves + calendar query